src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,148 @@
+        user.setAge(17);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("User must be at least 18 years old", thrown.getMessage());
+    }
+
+    @Test
+    public void register_ageExactly18_ok() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("qwertyQ123");
+        user.setAge(18);
+
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser, "User should be registered");
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+        assertEquals(user.getPassword(), registeredUser.getPassword());
+        assertEquals(user.getAge(), registeredUser.getAge());
+    }
+
+    @Test
+    public void register_duplicateLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("validUser123");
+        firstUser.setPassword("qwertyQ123");
+        firstUser.setAge(20);
+        registrationService.register(firstUser);
+
+        User secondUser = new User();
+        secondUser.setLogin("validUser123");
+        secondUser.setPassword("password456");
+        secondUser.setAge(22);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(secondUser));
+        assertEquals("User with this login already exists", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("qwertyQ123");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword(null);
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullAge_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("qwertyQ123");
+        user.setAge(null);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Age cannot be null", thrown.getMessage());
+    }
+
+    @Test
+    public void register_nullUser_notOk() {
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(null));
+        assertEquals("User cannot be null", thrown.getMessage());
```

Comment:
The test `register_nullUser_notOk` checks for a null user input, which is good. However, ensure that the corresponding `register` method in `RegistrationServiceImpl` throws a meaningful exception when a null user is passed to it.